cou/apps/base.py

```diff
@@ -160,6 +160,8 @@ def apt_source_codename(self) -> OpenStackRelease:
 
         if self.os_origin == "distro":
             # find the OpenStack release based on ubuntu series
+            if self.series not in DISTRO_TO_OPENSTACK_MAPPING:
+                raise ApplicationError(f"Series '{self.series}' is not supported by COU.")
```

Comment:
non-blocking nit: I wonder if it's good behavior having a property that could throw an exception if called. At the end of the day if the app class does not support a certain series we should probably throw an error earlier on in the constructor. And as a sidenote (out of scope here), I also wonder if it would have been a better design to push the distro and uca to release mapping within the OpenStackRelease class, to allow things like
```
OpenStackRelease('focal') # Ussuri
OpenStackRelease('cloud:focal-yoga') # Yoga
```

Reply:
I agree, it looks more like work for sanity check instead of property.